apps/saleor-app-checkout/backend/payments/providers/stripe/createPayment.ts

```diff
@@ -0,0 +1,136 @@
+          address: {
+            city: order.shippingAddress.city,
+            country: order.shippingAddress.country.code,
+            line1: order.shippingAddress.streetAddress1,
+            line2: order.shippingAddress.streetAddress2,
+            postal_code: order.shippingAddress.postalCode,
+            state: order.shippingAddress.countryArea,
+          },
+        }
+      : null,
+  });
+
+  const stripePaymentMethod = saleorPaymentMethodIdToStripePaymentMethodId(method);
+
+  order.discounts;
+
+  const stripeCheckoutSession = await stripeClient.checkout.sessions.create({
+    line_items: [
+      ...order.lines.map(saleorLineToStripeLine),
+      ...order.discounts.map(saleorDiscountToStripeLine),
+      saleorOrderShippingToStripeLine(order),
+    ],
+
+    // @todo
+    locale: "en",
+
+    payment_method_types: stripePaymentMethod ? [stripePaymentMethod] : undefined,
+    customer: stripeCheckoutCustomer.id,
+    mode: "payment",
+    cancel_url: formatRedirectUrl(redirectUrl, order.id),
+    success_url: formatRedirectUrl(redirectUrl, order.id),
+    metadata: {
+      orderId: order.id,
+    },
+  });
+
+  return {
+    url: stripeCheckoutSession.url,
+    id: stripeCheckoutSession.id,
+  };
+};
+
+type SaleorLine = OrderFragment["lines"][number];
+const saleorLineToStripeLine = (line: SaleorLine): Stripe.Checkout.SessionCreateParams.LineItem => {
+  return {
+    price_data: {
+      currency: line.unitPrice.gross.currency.toUpperCase(),
+      unit_amount: getIntegerAmountFromSaleor(line.unitPrice.gross.amount),
+      product_data: {
+        name: line.productName + "-" + line.variantName,
+        images: line.thumbnail?.url ? [line.thumbnail.url] : [],
+      },
+    },
+    quantity: line.quantity,
+  };
+};
+type SaleorDiscount = OrderFragment["discounts"][number];
+const saleorDiscountToStripeLine = (
+  discount: SaleorDiscount
+): Stripe.Checkout.SessionCreateParams.LineItem => {
+  return {
+    price_data: {
+      currency: discount.amount.currency.toUpperCase(),
+      unit_amount: getIntegerAmountFromSaleor(discount.amount.amount),
+      product_data: {
+        name: "Discount " + (discount.name || ""),
+        images: [],
+      },
+    },
+    quantity: 1,
+  };
+};
+
+const saleorOrderShippingToStripeLine = (
+  order: OrderFragment
+): Stripe.Checkout.SessionCreateParams.LineItem => {
+  return {
+    quantity: 1,
+    price_data: {
+      currency: order.shippingPrice.gross.currency.toUpperCase(),
+      unit_amount: getIntegerAmountFromSaleor(order.shippingPrice.gross.amount),
+      product_data: {
+        name: "Shipping " + (order.shippingMethodName || ""),
+        images: [],
+      },
+    },
+  };
+};
+
+const saleorPaymentMethodIdToStripePaymentMethodId = (
+  paymentMethodId: PaymentMethodID
+): Stripe.Checkout.SessionCreateParams.PaymentMethodType | null => {
+  switch (paymentMethodId) {
+    case "creditCard":
+      return "card";
+    case "applePay":
+      // @todo ?
```

Comment:
These will be solved in another pr or..?

Reply:
I don't know yet how to tackle this. It doesn't look like Stripe currently offers a way to do Apple Pay with the payment flow we're using.

Reply:
So maybe let's create a ticket for that in the project and link the ticket in the comment there? That way in half a year when we see that todo we'll know the reason this wasn't done in the first place

Reply:
https://github.com/saleor/react-storefront/issues/390